cordova-serve/src/browser.js

```diff
@@ -95,16 +105,93 @@ function getBrowser(target, dataDir) {
             'firefox': 'firefox',
             'opera': 'opera'
         },
-        'linux' : {
-            'chrome': 'google-chrome' + chromeArgs ,
+        'linux': {
+            'chrome': 'google-chrome' + chromeArgs,
             'chromium': 'chromium-browser' + chromeArgs,
             'firefox': 'firefox',
             'opera': 'opera'
         }
     };
-    target = target.toLowerCase();
     if (target in browsers[process.platform]) {
-        return Q(browsers[process.platform][target]);
+        var browser = browsers[process.platform][target];
+        if (process.platform === 'win32') {
+            // Windows displays a dialog if the browser is not installed. We'd prefer to avoid that.
+            return checkBrowserExistsWindows(browser, target).then(function () {
+                return browser;
+            });
+        } else {
+            return Q(browser);
+        }
     }
-    return Q.reject('Browser target not supported: ' + target);
+    return Q.reject(NOT_SUPPORTED.replace('%target%', target));
+}
+
+function checkBrowserExistsWindows(browser, target) {
+    var promise = target === 'edge' ? edgeSupported() : browserInstalled(browser);
+    return promise.catch(function (error) {
+        return Q.reject((error && error.toString() || NOT_INSTALLED).replace('%target%', target));
+    });
+}
+
+function edgeSupported() {
+    var d = Q.defer();
+
+    child_process.exec('ver', function (err, stdout, stderr) {
```

Comment:
Is there any chance to reuse `./exec` module here. Looks like it contains almost the same logic.